.github/actions/fetch-artifact/action.yml

```diff
@@ -0,0 +1,63 @@
+name: Fetch artifact
+description: Fetches a specific artifact and downloads it to the current directory
+
+inputs:
+  name:
+    description: The name of the artifact to fetch
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: find_release_artifact
+      id: find_release_artifact
+      uses: actions/github-script@v7
+      with:
+        result-encoding: string
+        script: | # js
+          const fs = require('fs');
+
+          const artifacts = await github.rest.actions.listArtifactsForRepo({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            name: "${{ inputs.name }}",
+            per_page: 1,
+          });
+
+          if (!artifacts.data?.artifacts?.[0]?.id) {
+            throw new Error(`No artifacts found for ${{ inputs.commit }}`);
+          }
+
+          const artifact_id = artifacts.data.artifacts[0].id;
+
+          const download = await github.rest.actions.downloadArtifact({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            artifact_id: artifact_id,
+            archive_format: 'zip',
+          });
+
+          fs.writeFileSync(`${process.env.GITHUB_WORKSPACE}/mb.zip`, Buffer.from(download.data));
+
+    - name: unzip uberjar artifact
+      shell: bash
+      run: unzip mb.zip
+    - name: Move the Uberjar
+      shell: bash
+      run: | # bash
+        if [ -e "metabase.jar" ]; then
+          echo "metabase.jar is already in the right place"
+        elif [ -e "target/uberjar/metabase.jar" ]; then
+            echo "moving metabase.jar to current directory"
+          mv target/uberjar/metabase.jar metabase.jar
+        else
+          echo "Could not find the metabase.jar file"
+          exit 1
+        fi
+
+    - name: Verify that this is a valid JAR file
+      shell: bash
+      run: file --mime-type ./metabase.jar
```

Comment:
fun fact:
- our build process spits out a jar with `application/zip` mime type
- since we use java to process the jar when editing the version.properties, it changes to `application/java-archive`